### PR TITLE
Ret 1766

### PR DESCRIPTION
**Pull request by vinothon84**
1) API service for to send email
2) Junit tests included
3) yaml file updated with api key and sample template id
4) Read.md section updated for pact

### CODE REVIEW

**Review by nickwest1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/exception/NotificationException.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmcts.reform.et.syaapi.exception;
+
+public class NotificationException extends RuntimeException {
+
+    private static final long serialVersionUID = -3042681110164047L;
```
Comment (Brogie) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/exception/NotificationException.java:5]: What is the reason for such a specific number for this? is this the minimum value for long? If so you can use long.min_value for easier to read code.

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/notification/NotificationsProperties.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+@Validated
+@Data
+public class NotificationsProperties {
+
+    @Value("${notifications.govNotifyApiKey}")
+    private String govNotifyApiKey;
+
+    @Value("${notifications.sampleEmailTemplateId}")
```
Comment (Brogie) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/notification/NotificationsProperties.java:18]: Does this need to be extendable for future use? maybe need some form of lookup for different templates

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+     * Given a caseId, initialization of trigger event to start and submit update for case.
+     *
+     * @param templateId - pass gov notify template id for each use case
+     * @param targetEmail - recepient target email id
+     * @param parameters - map of strings to add this to the template
+     * @param reference - reference string for email template
+     * @return response from notification api
+     */
+
+    public SendEmailResponse sendEMail(
```
Comment (Brogie) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:33]: inconsistant method capitalisation sendEMail here and sendEmail in the notification client, sendEmail is prefered.

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -50,4 +50,8 @@ acas:
 tornado:
   api:
     url: ${TORNADO_URL:http://localhost:8090/rs/render}
-    key: ${TORNADO_ACCESS_KEY:ignored_default_value}
\ No newline at end of file
+    key: ${TORNADO_ACCESS_KEY:ignored_default_value}
+
+notifications:
+  govNotifyApiKey: ${gov-notify-api-key:}
+  sampleEmailTemplateId: "4ea3cf16-0cbb-4f27-8cfb-a084b126086e"
```
Comment (Brogie) [inline src/main/resources/application.yaml:57]: Is this where we could extend for other templates by adding new template ID's here?

File: src/test/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.doReturn;
+
+@SuppressWarnings({"PMD.TooManyMethods"})
+class NotificationServiceTest {
+    private static final String TEST_GOV_NOTIFY_API_KEY =
+        "test_test_key-002d2170-e381-4545-8251-5e87dab724e7-190d8b02-2bb8-4fc9-a471-5486b77782c0";
```
Comment (Brogie) [inline src/test/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationServiceTest.java:24]: Are these ok to commit?

File: src/test/java/uk/gov/hmcts/reform/et/syaapi/utils/TestConstants.java [hunk 6]
```diff
@@ -8,6 +8,7 @@ public final class TestConstants {
     public static final String DRAFT = "DRAFT";
     public static final String AWAITING_SUBMISSION_TO_HMCTS = "AWAITING_SUBMISSION_TO_HMCTS";
     public static final String SUBMITTED = "SUBMITTED";
+    public static final String TEST_STRING = "TEST";
 
     private TestConstants() {
         // Utility clasa
```
Comment (Brogie) [inline src/test/java/uk/gov/hmcts/reform/et/syaapi/utils/TestConstants.java:14]: // Utility class

**Review by Brogie — COMMENTED** [review]
Mostly questions that can be ignored in this review (Maybe chat through on call), but some changes needed around method names to keep them consistant with the rest of the project.

File: src/test/java/uk/gov/hmcts/reform/et/syaapi/utils/TestConstants.java (hunk 6 above)
Comment (vinothon84) [reply src/test/java/uk/gov/hmcts/reform/et/syaapi/utils/TestConstants.java:14]: Let me remove the comment.

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java (hunk 3 above)
Comment (vinothon84) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:33]: well spotted and let me update this.

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/exception/NotificationException.java (hunk 1 above)
Comment (vinothon84) [reply src/main/java/uk/gov/hmcts/reform/et/syaapi/exception/NotificationException.java:5]: Let me try updating as per your comment.

File: src/test/java/uk/gov/hmcts/reform/et/syaapi/utils/TestConstants.java (hunk 6 above)
Comment (vinothon84) [reply src/test/java/uk/gov/hmcts/reform/et/syaapi/utils/TestConstants.java]: Cannot remove this as check style was complaining.

**Review by pmumbere — APPROVED** [review]

**Pull request merged**